lambda/matching.ml

```diff
@@ -2947,8 +2947,14 @@ let combine_variant loc row arg partial ctx def (tag_lambda_list, total1, _pats)
         | [ (_, act1) ], [ (_, act2) ] when fail = None ->
             test_int_or_block arg act1 act2
         | _, [] ->
-            (* One can compare integers and pointers *)
-            make_test_sequence_variant_constant fail arg consts
+            begin match fail with
+            | None ->
+              make_test_sequence_variant_constant fail arg consts
+            | Some act ->
+              test_int_or_block arg
+                (make_test_sequence_variant_constant fail arg consts)
+                act
+            end
```

Comment:
I think the code would be nicer if you used the same style as the case below, which (if I understand correctly) is exactly symmetric.

Reply:
I've followed your suggestion. The code now follows the style of the case below.